src/main/webapp/app/course/course-scores/course-scores.component.ts

```diff
@@ -144,14 +144,17 @@ export class CourseScoresComponent implements OnInit, OnDestroy {
             }
 
             // find all students by iterating through the participations
-            let student = studentsMap.get(participation.student.id!);
-            if (student == null) {
-                student = new Student(participation.student);
-                studentsMap.set(participation.student.id!, student);
-            }
-            student.participations.push(participation);
-            if (participation.presentationScore) {
-                student.presentationScore += participation.presentationScore;
+            const students = participation.student ? [participation.student] : participation.team.students;
+            for (let i = 0; i < students.length; i++) {
```

Comment:
can we use a for each loop like here, something like the following?
(const student of students)

By the way, we should always prefer for each loops if they can be used

Reply:
One more question: are you sure this is enough code so that really all students in the team get the points for their final score? Also in a mixed scenario with some individual and some team participations?

Reply:
I switched the loop to a `for (const x of xs)` construct.

Yes, I just verified that this works as expected. In this scenario, the three marked students belong to the same team for a team-based text exercise. They received a result of 30/100 points for their submission. It was correctly credited to all of them and all of them were considered to have participated in this exercise.

![OnPaste 20200329-134143](https://user-images.githubusercontent.com/7109225/77848161-303bac80-71c3-11ea-9d4c-13dfe1aa6794.png)